docs/specifications/tests/Nameserver-TP/nameserver11.md

```diff
@@ -0,0 +1,84 @@
+# NAMESERVER11: Test for undefined EDNS OPTION-CODE
+
+## Test case identifier
+
+**NAMESERVER11** 
+
+
+## Objective
+
+EDNS is a mechanism to announce capabilities of a DNS implementation,
+and is now basically required by any new functionality in DNS such as
+DNSSEC ([RFC 6891]).
+
+[RFC 6891, section 6.1.2] states that any OPTION-CODE values not understood by a
+responder or requestor MUST be ignored. Unknown OPTION-CODE values must be
+processed as though the OPTION-CODE was not even there.
+
+In this test case, we will query  with an unknown EDNS OPTION-CODE and expect
+that the OPTION-CODE is not present in the response for the query.
+
+## Inputs
+
+"Child Zone" - The domain name to be tested.
+
+## Ordered description of steps to be taken to execute the test case
+
+1. Create a SOA query for the *Child Zone* with an OPT record with 
+   EDNS OPTION-CODE set to anything other than it is already assigned as in the
+[IANA-DNSSYSTEM-PARAMETERS] and no other EDNS options or flags.
+
+2. Obtain the set of name server IP addresses using [Method4] and [Method5]
+   ("Name Server IP").
+
+3. For each name server in *Name Server IP* do:
+	1. Send the SOA query to the name server and collect the response.
+	2. If there is no DNS response, output *[NO_RESPONSE]* and go to
+	next server.
+	3. Else, if the DNS response has the RCODE "FORMERR" then output
+      	*[NO_EDNS_SUPPORT]*.
+	4. Else, if the DNS response meet the following four criteria,
+      	then just go to the next name server (no error):
+		1. The SOA is obtained as response in the ANSWER section.
+		2. If the DNS response has the RCODE "NOERROR".
+		3. The pseudo-section response has an OPT record with version set to 0.
+		4. There is no "OPTION-CODE" present in the response.
+
```

Comment:
I think it makes sense to have a check if the unassigned OPTION-CODE that we set in the query is there in the response. If so send a specific message since that is what we test for, INVALID_OPTION_CODE.

Or maybe check for any "invalid" OPTION-CODE?

Compare with the check we have in DNSCHECK on valid algorithms.

Reply:
Updated